Fixes CSV import of hierarchical brands

PanosSynetos opened the pull request:
Addresses an issue where brands with parent-child
relationships were not being imported correctly
during CSV product imports.

This change ensures proper parsing of hierarchical brand names during the import process, resolving the import issue.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Before #61366 formatting was happening earlier in the process and `>` was already coming `&gt;`.
The `>` is used to identify hierarchical brands. I've updated the brands code, to harmonize it with how the categories re imported.

Closes wooplug-5804
Closes #61823

(For Bug Fixes) Bug introduced in PR #61366
- #61366

### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

| Before | After |
| ------ | ----- |
|  <img width="1992" height="662" alt="image" src="https://github.com/user-attachments/assets/fd190797-a5ee-46f5-bc8a-e810628a99bd" />    |   <img width="1948" height="665" alt="image" src="https://github.com/user-attachments/assets/f778335f-d22a-4703-bc71-affe73b76fd7" />    |
|  <img width="600" height="337" alt="image" src="https://github.com/user-attachments/assets/721776ec-f97d-410a-af89-4de70e98c4b3" />    |   <img width="618" height="342" alt="image" src="https://github.com/user-attachments/assets/9aa3663e-c760-44d5-89a8-fa2f251c2443" />    |

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Use this CSV

```
SKU,Brands,Categories
BP1,TopBrand,TopBrand,
BP2,TopBrand > KidCakes,TopBrand > KidCakes,
BP3,"TopBrand, TopBrand > KidCakes","TopBrand, TopBrand > KidCakes",
BP4,"TopBrand, TopBrand > Slice","TopBrand, TopBrand > Slice",
```

1. Go to Products > All Products > Import and upload the CSV
2. Make sure that the mapping is correct
3. Import the products
4. Go to Products > All Products and see that Brands are correctly associated with the products
5. Go to Products > Brands and verify that Brands are inserted correctly and are hierarchical

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

PanosSynetos committed:
Fixes CSV import of hierarchical brands

Addresses an issue where brands with parent-child
relationships were not being imported correctly
during CSV product imports.

This change ensures proper parsing of hierarchical brand
names during the import process, resolving the import issue.

Closes wooplug-5804
Closes #61823

PanosSynetos committed:
Fixes typo

PanosSynetos commented on plugins/woocommerce/includes/admin/class-wc-admin-brands.php:
```diff
@@ -746,12 +750,15 @@ public function parse_brands_field( $value ) {
+				// Don't allow users without capabilities to create new categories.
+				if ( ! current_user_can( 'manage_product_terms' ) ) {
+					break;
+				}
```
I've pushed a commit to fix the "new categories" vs "new brands". As for the logging, I've used the same process as we use in the categories importer

xristos3490 reviewed (approved):
Looking good and tests well on my end!

Before:
<img width="743" height="442" alt="Screenshot 2025-11-06 at 1 47 57 PM" src="https://github.com/user-attachments/assets/89a2a7dc-551d-4727-b45e-2d2510eab763" />

After:
<img width="891" height="420" alt="Screenshot 2025-11-06 at 1 49 08 PM" src="https://github.com/user-attachments/assets/6ed458a9-20ab-4486-bd1c-b6df1baa3f40" />

PanosSynetos committed:
First take in unit testing brands import

PanosSynetos committed:
Adds test for product importer brand handling

Adds a test case to verify that the product CSV importer correctly handles product brands, including hierarchical relationships.

This test ensures that brands are properly assigned to products during import, addressing an issue where child brands were not being correctly imported.

PanosSynetos committed:
Merge branch 'trunk' into wooplug-5804-child-brands-not-imported-properly-during-core-product

PanosSynetos committed:
Make PHPCS happy

PanosSynetos committed:
Merge remote-tracking branch 'origin/wooplug-5804-child-brands-not-imported-properly-during-core-product' into wooplug-5804-child-brands-not-imported-properly-during-core-product

PanosSynetos committed:
Merge branch 'trunk' into wooplug-5804-child-brands-not-imported-properly-during-core-product

PanosSynetos committed:
Initializes brands

PanosSynetos committed:
Require brands files

PanosSynetos committed:
CI tests fail - trying to fix them

PanosSynetos committed:
Skip test_get_parsed_data_brands

PanosSynetos committed:
Debug CI

PanosSynetos committed:
Loads brands admin and class unconditionally.

PanosSynetos committed:
Trying to make CI happy

PanosSynetos committed:
Make CI happy

PanosSynetos committed:
Some more

PanosSynetos committed:
Let's see now

PanosSynetos committed:
Nailed it?

xristos3490 reviewed (approved):
Looking good! Thanks for adding the unit test here! 👌

PanosSynetos commented on plugins/woocommerce/tests/legacy/unit-tests/importer/product.php:
```diff
@@ -38,6 +38,17 @@ public function setUp(): void {
+
+		require_once $bootstrap->plugin_dir . '/includes/admin/class-wc-admin-brands.php';
+		$this->brands_admin = new WC_Brands_Admin();
+
```
I've fixed this in the next commit https://github.com/woocommerce/woocommerce/pull/61834/commits/fe73377111ae68b30dd7af97f95b682d6bdae7ef
